src/Consolonia.Core/Infrastructure/ConsoleCursorFactory.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using Avalonia;
+using Avalonia.Input;
+using Avalonia.Platform;
+
+namespace Consolonia.Core.Infrastructure
+{
+#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
+    internal sealed class ConsoleCursorFactory : ICursorFactory
+    {
+        private readonly Dictionary<StandardCursorType, CursorImpl> _standardCursors = new();
+
+        public ICursorImpl GetCursor(StandardCursorType cursorType)
+        {
+            lock (_standardCursors)
+            {
+                if (_standardCursors.TryGetValue(cursorType, out CursorImpl cursor))
+                    return cursor;
+                cursor = new CursorImpl(cursorType);
+                _standardCursors.Add(cursorType, cursor);
+                return cursor;
+            }
+        }
+
+        public ICursorImpl CreateCursor(IBitmapImpl cursor, PixelPoint hotSpot)
+        {
+            throw new NotSupportedException("Consolonia doesn't support bitmap based cursors");
```

Comment:
I think it's a good place to RaiseNotSupported. But again, we need to improve it for possibility to return things, not just throwing exceptions.It's just for future discussions